pkg/cmd/admin/node/listpods.go

```diff
@@ -0,0 +1,101 @@
+package node
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	kerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+)
+
+type ListPodsOptions struct {
+	Options *NodeOptions
+}
+
+func (l *ListPodsOptions) AddFlags(cmd *cobra.Command) {
+	kcmdutil.AddPrinterFlags(cmd)
+}
+
+func (l *ListPodsOptions) Run() error {
+	r, err := l.Options.GetNodeResource()
+	if err != nil {
+		return err
+	}
+
+	errList := []error{}
+	nodeCount := 0
+	err = r.Visit(func(info *resource.Info) error {
+		nodeCount++
+		node, ok := info.Object.(*kapi.Node)
+		if !ok {
+			err = fmt.Errorf("cannot convert input to Node: ", reflect.TypeOf(info.Object))
+			errList = append(errList, err)
+			// Don't bail out if one node fails
+			return nil
+		}
+
+		err := l.RunListPods(node)
+		if err != nil {
+			errList = append(errList, err)
+			// Don't bail out if one node fails
+			return nil
+		}
+		return nil
+	})
+	if len(errList) != 0 {
+		return kerrors.NewAggregate(errList)
+	}
+	if nodeCount == 0 {
+		if len(l.Options.NodeNames) > 0 {
```

Comment:
Same comment as above.  You can run the check based on `util.StringSet.Difference()`.